src/utils/Utils.ts

```diff
@@ -438,4 +438,16 @@ export class Utils {
   static resolveAfter<T = void>(ms: number, returns?: T): Promise<T> {
     return new Promise(resolve => setTimeout(() => (returns !== undefined ? resolve(returns) : resolve()), ms));
   }
+
+  static reorderValuesByKeys<T, K extends string>(values: T[], order: K[], getKey: (value: T) => K, appendRemainingValues = true) {
```

Comment:
I added the option `appendRemainingValues` since I felt like it clarified what the function does.

Reply:
Is the `appendRemainingValues` used in this PR? I don't see where it is being passed as `false` outside of tests.
I think the ternary on line 451 is not needed, since the `else` path is never triggered by the source code.

Reply:
As mentioned above, it was to clarify what this function does, since I wouldn't normally expect a "reorder" function to include unordered values.

I'll remove it, though.